### PR TITLE
thuang-feedback-1062

### PR DESCRIPTION
**Pull request by tihuan**
### Reviewers
**Functional:** 
@seve 

**Readability:** 

---

## Changes
1. Update `removeParams()` function to use web API more
2. Add collection `CURATOR_FEATURES` to host all feature flags we want to enable for curators
3. Split featureFlags/index.ts to have getter and setter (`checkFeatureFlags()`)
4. Check for feature flags at app initiation time `checkFeatureFlags()` in `_app.tsx`. The setter will NOT set `rubber-chicken` to `true` in localStorage, so other feature flags remain the single source of truth. This is to prevent the weird state where in localStorage `cc=true` and `rubber-chicken=false`
5. Remove removing the feature flag params, since getter and setter are now separated

## Definition of Done (from ticket)
1. feature flags should work -- TESTED
2. Redirect after login should work -- NOT TESTED PLEASE HELP 😆 

## Known Issues
N/A

### CODE REVIEW

File: frontend/src/common/featureFlags/features.ts [hunk 1]
```diff
@@ -4,3 +4,5 @@ export enum FEATURES {
   GENE_SETS = "gs",
   CURATOR = "rubber-chicken",
 }
+
+export const CURATOR_FEATURES = [FEATURES.CREATE_COLLECTION, FEATURES.AUTH];
```
Comment (seve) [inline frontend/src/common/featureFlags/features.ts:8]: We're planning to deprecate the cc and auth flags, which is why I had them both in code for the moment while Brian communicates the deprecation

File: frontend/src/common/featureFlags/index.ts [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
+    }
+  });
+}
+
+function setFeatureFlag(key: string, value: string) {
+  const URLValueAsBooleanString =
+    value === "true" ? BOOLEAN.TRUE : BOOLEAN.FALSE;
+
+  setLocalStorage(FEATURE_FLAG_PREFIX + key, URLValueAsBooleanString);
+}
```
Comment (seve) [inline frontend/src/common/featureFlags/index.ts:39]: There's no check to ee if the key is a valid feature flag, don't we risk dirtying the local storage?

File: frontend/src/components/CreateCollectionModal/index.tsx [hunk 3]
```diff
@@ -41,10 +40,7 @@ const CreateCollection: FC<{
   const [isOpen, setIsOpen] = useState(shouldModuleOpen);
```
Comment (seve) [inline frontend/src/components/CreateCollectionModal/index.tsx:40]: Turns out, this has been broken since #1016

Comment (seve) [reply frontend/src/components/CreateCollectionModal/index.tsx:40]: We need to define a new BOOLEAN enum. https://github.com/chanzuckerberg/corpora-data-portal/pull/762#pullrequestreview-546520583

**Review by seve — CHANGES_REQUESTED** [review]

File: frontend/src/components/CreateCollectionModal/index.tsx (hunk 3 above)
Comment (tihuan) [reply frontend/src/components/CreateCollectionModal/index.tsx:40]: ahh great catch! I changed it to `href={`${API_URL}${API.LOG_IN}?redirect=?${QUERY_PARAMETERS.LOGIN_MODULE_REDIRECT}=${BOOLEAN.TRUE}`}`!

File: frontend/src/common/featureFlags/features.ts (hunk 1 above)
Comment (tihuan) [reply frontend/src/common/featureFlags/features.ts:8]: Ooh I see! I'll add the checks back at callsites 👍

File: frontend/src/components/CreateCollectionModal/components/CTA/index.tsx [hunk 4]
```diff
@@ -28,7 +29,7 @@ const CTA: FC<Props> = ({ onClose }) => {
           </Button>
           <AnchorButton
             intent={Intent.PRIMARY}
-            href={`${API_URL}${API.LOG_IN}?redirect=?${QUERY_PARAMETERS.LOGIN_MODULE_REDIRECT}=true`}
+            href={`${API_URL}${API.LOG_IN}?redirect=?${QUERY_PARAMETERS.LOGIN_MODULE_REDIRECT}=${BOOLEAN.TRUE}`}
```
Comment (tihuan) [inline frontend/src/components/CreateCollectionModal/components/CTA/index.tsx:32]: use `BOOLEAN.TRUE` instead of hard coded `true`

**Comment by tihuan** [comment]
@seve just updated the PR! PTAL again thank you!!

**Pull request merged**